themes/digital.gov/src/js/github.js

```diff
@@ -1,89 +1,158 @@
-// Disabling no-undef because some vars are set in templates.
-// @TODO: Remove var setting in Hugo templates.
-/* eslint-disable no-undef */
-
-function getBranchLink(branch) {
-  const path = `https://github.com/GSA/digitalgov.gov/tree/${branch}`;
-  const branchLink = `<a class="branch" href="${path}" title="${branch}">${branch}</a>`;
-  return branchLink;
-}
-
-function formatDate(timezoneDate) {
-  const inputDate = new Date(timezoneDate);
-
-  const dateOptions = {
-    day: "numeric",
-    month: "short",
-    year: "numeric",
-  };
-
-  const timeOptions = {
-    hour: "numeric",
-    minute: "2-digit",
-    timeZone: "America/New_York",
-    timeZoneName: "shortGeneric",
-  };
-
-  const outputDate = inputDate.toLocaleDateString(undefined, dateOptions);
-  const outputTime = inputDate
-    .toLocaleTimeString("en-US", timeOptions)
-    .replace("AM", "a.m.,")
-    .replace("PM", "p.m.,");
-  return `${outputDate} at ${outputTime}`;
-}
-
-function showLastCommit(data, branch) {
-  const branchLink = getBranchLink(branch);
-  const commitData = Array.isArray(data) ? data[0] : data;
-  const commitDate = commitData.commit.committer.date;
-  // eslint-disable-next-line no-undef
-  const commitHistoryUrl = `https://github.com/GSA/digitalgov.gov/commits/${branch}/content/${filepath}`;
-
-  const lastCommit = [
-    branchLink,
-    `<p>Last updated on <a href=${commitHistoryUrl}>
-        <span class='commit-date'>${formatDate(commitDate)}</span>
-      </a></p>`,
-  ];
-
-  // eslint-disable-next-line no-unused-vars, func-names
-  $(".edit-file").each(function (i, itemsList) {
-    $(this).append(lastCommit.join("\n"));
-  });
-}
-
-jQuery(($) => {
-  function buildEditFileLink() {
-    // editpathURL is set the <head>
-    if (editpathURL !== undefined) {
-      // Build the edit link
-      const edit = `<a target='_blank' class='edit-file-link' href='${editpathURL}' title='Edit in GitHub'>Edit</a>`;
-
-      // Insert the .edit-file-link html into the .edit-file div and remove the .hidden class
-      $("#page-data .edit-file").html(edit).removeClass("hidden");
+/**
+ * Sets the most recent commit timestamp for the page
+ * This is displayed in #page-data section at the bottom of the page
+ */
+
+// eslint-disable-next-line func-names
+(function () {
+  const editFileButton = document.querySelector(".edit-file");
+  const editPageLink = document.querySelector("#page-data .edit-file");
+  const gitRepo = {};
```

Comment:
We should export the shared settings to their own JS file and import that data in each script.

>**Warning**
> This functionality is added in #6419. So we can leave for now and refactor in the future.